spec/Spec/BlsEC/FQP.cry

```diff
@@ -0,0 +1,328 @@
+/**
+ * This module provides support for polynomial extension fields.
+ * Specifically, we need to support Fq2 and Fq12.
+ * @see https://github.com/ethereum/py_ecc/blob/70c194edd8d1eb1457805442f0162499dbc0aac5/py_ecc/fields/field_elements.py#L204
+ */
+module Spec::BlsEC::FQP where
+
+import Spec::BlsEC::Field
+import Spec::BlsEC::Poly(poly_extended_euclidean)
+
+/**
+ * A polynomial field extension type contains a list of
+ *  coefficients which are elements of Fq.
+ */
+type FQP n = [n](Z Fq)
+
+/**
+ * A G2 point is comprised of two Fq2 elements, or sets of coefficients, each of degree two.
+ * Each element is comprised of a real and imaginary coeffsicient.
+ */
+type Fq2 = FQP 2
```

Comment:
https://github.com/GaloisInc/ckzg-eip-4844-verification/blob/633cc70583bb6c6c195e8770e9d9fa3d3eabc13b/spec/Spec/BlsEC/G2.cry#L52

Line 52 was broken by this change from an enum to to Fq2 it still has the real and imaginary bit

Reply:
Also broke this test in Poly.cry
https://github.com/GaloisInc/ckzg-eip-4844-verification/blob/633cc70583bb6c6c195e8770e9d9fa3d3eabc13b/spec/Spec/BlsEC/Poly.cry#L15